go/vt/discovery/healthcheck.go

```diff
@@ -293,39 +299,42 @@ type HealthCheck interface {
 // about a tablet.
 type healthCheckConn struct {
 	// set at construction time
-	ctx        context.Context
-	cancelFunc context.CancelFunc
+	ctx context.Context
 
 	// mu protects all the following fields.
 	// When locking both mutex from HealthCheck and healthCheckConn,
 	// HealthCheck.mu goes first.
 	// Note tabletStats.Tablet and tabletStats.Name are immutable.
-	mu                    sync.RWMutex
 	conn                  queryservice.QueryService
-	streamCancelFunc      context.CancelFunc
 	tabletStats           TabletStats
 	loggedServingState    bool
 	lastResponseTimestamp time.Time // timestamp of the last healthcheck response
 }
 
+// tabletHealth maintains the health status of a tablet fed by the health check connection.
+type tabletHealth struct {
+	cancelFunc  context.CancelFunc
```

Comment:
It would be helpful to have some comments indicating what each of these fields does. In particular I think this used to be called `streamCancelFunc` which made it (somewhat) clearer what it did.

Also it's not clear from an initial read what the relationship is between the `tabletStats` in this struct vs the one in healthCheckConn.

IMO renaming to something like `latestTabletStats` would make it clearer.